letsencrypt: Update certbot and dns plugins

GoetzGoerisch opened the pull request:
## Fixed
- addresses GHSA-79v4-65xg-pq4g

## Breaking
- Disabled certbot-dns-mijn-host
  - issue: [mijnhost/certbot-dns-mijn-host#8](https://github.com/mijnhost/certbot-dns-mijn-host/issues/8)
- Disabled certbot-dns-websupport
  - issue: [johnybx/certbot-dns-websupport#1](https://github.com/johnybx/certbot-dns-websupport/issues/1)

## Updated
- Update certbot to 3.1.0
- Update acme to 3.1.0
- Update cryptography to 44.0.1
- Update certbot-dns-azure to 2.6.1
- Update certbot-dns-directadmin to 10.0.13
- Update certbot-dns-duckdns to 1.5
- Update certbot-dns-dynu to 0.0.6
- Update certbot-dns-inwx to 3.0.1
- Update certbot-dns-ionos to 2024.11.9
- Update certbot-dns-norisnetwork to 0.3.0
- Update certbot-dns-porkbun to 0.9.1
- Update certbot-dns-netcup to 1.4.4
- Update certbot-dns-njalla to 2.0.2
- Update the remaining dns challenges with DNS_PROVIDER placeholder

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Dependency Updates**
	- Updated multiple DNS provider and certificate management dependencies to their latest versions.
	- Upgraded Certbot and ACME to version 3.1.0.
	- Updated Cryptography and various DNS plugin versions.

- **Improvements**
	- Enhanced DNS provider argument handling for more flexible configuration.
	- Simplified INWX DNS service URL for improved access.
	- Added placeholder for remaining DNS challenges.
	- Updated version in the configuration file to 5.3.0.
	- Disabled support for `certbot-dns-mijn-host` and `certbot-dns-websupport` plugins due to unresolved issues.
	- Updated documentation to reflect the current status of disabled DNS providers.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

agners commented on letsencrypt/CHANGELOG.md:
```diff
@@ -1,5 +1,22 @@
 # Changelog
 
+## 5.2.11
```
Hm, I'd suggest to make this a minor bump (5.3.0), just to be on the safe side.

woodsb02 commented:
Heads up that the Porkbun provider is broken, and needs this update to work again.

This is because Porkbun recently changed the URL for their API:
Old Value: porkbun.com
New Value: api.porkbun.com
Deadline: 2024-12-01 00:00:00 UTC

This can be fixed by updating the addon to use at least the 0.9.0 version of certbot-dns-porkbun
https://github.com/infinityofspace/certbot_dns_porkbun/releases/tag/v0.9

The newest 0.9.1 release as proposed is the best bet.

frenck reviewed (changes requested):
Oh hi there @GoetzGoerisch 👋 

There is a merge conflict, could you please take a look?

Thanks 👍 

../Frenck

agners reviewed (changes requested):
Since this breaks at least two DNS validation providers let's add the `breaking_versions` configuration to config.yaml (see https://developers.home-assistant.io/docs/add-ons/configuration). This will prevent auto-update so folks with those DNS validation providers can opt to not (yet) update to the latest version.

GoetzGoerisch commented:
Thank you @angers for the review and the suggestion with the breaking versions,
 
> Since this breaks at least two DNS validation providers let's add the `breaking_versions` configuration to config.yaml (see https://developers.home-assistant.io/docs/add-ons/configuration). This will prevent auto-update so folks with those DNS validation providers can opt to not (yet) update to the latest version.

Fixed with [e3ccb12](https://github.com/home-assistant/addons/pull/3902/commits/e3ccb124f70220df2909912cca3c41e1ecfd8025)

agners commented on letsencrypt/config.yaml:
```diff
@@ -1,5 +1,6 @@
 ---
 version: 5.3.0
+breaking_versions: 5.3.0
```
This sho

agners commented on letsencrypt/config.yaml:
```diff
@@ -1,5 +1,6 @@
 ---
-version: 5.2.12
+version: 5.3.0
+breaking_versions: [ 5.3.0 ]
```
```suggestion
breaking_versions: [5.3.0]
```

GoetzGoerisch commented on letsencrypt/CHANGELOG.md:
```diff
@@ -1,5 +1,28 @@
+- Update certbot-dns-porkbun to 0.9.1
+- Update certbot-dns-netcup to 1.4.4
+- Update certbot-dns-njalla to 2.0.2
+- Update the remaining dns challenges with DNS_PROVIDER placeholder
```
https://github.com/home-assistant/addons/pull/3902/commits/8718ea96d9060662ff7ad6011a4f3fc55e09556a

GoetzGoerisch replied:
https://github.com/home-assistant/addons/pull/3902/commits/4b2bab25a3747e8173a224ff0ac7078218a087b9

GoetzGoerisch replied:
@coderabbitai show me the PROVIDER_ARGUMENTS location you are referring to.

GoetzGoerisch commented:
@agners thank you for the fix I have incorporated it into the correct commit attributing you as the co-author.
I also update cryptography to address the published CVE. Futhermore, I fixed another suggestion by coderabbit.

Could you please give the CI another go?
